src/test/unit/app/controller/courts/AddressController.ts

```diff
@@ -0,0 +1,466 @@
+import {mockRequest} from '../../../utils/mockRequest';
```

Comment:
Made some notes of my own for unit tests, can you check that these are here? 

```
GET/render
	

	Success: 

		Gets addresses x 

		Gets address types x

		getAddressTypesForSelect

			primary address returns all 

			secondary address returns all matching allAddressTypes variable 

	Failure: 

		get addresses - error message 

		get address types - error message 

PUT 


	CSRF token false 

	validateAddress - 

		validate type

			type required error 

		
		validate address lines 

			address required 

			town required 

		
		validate postcode 

			postcode empty 

			invalid postcode format


	validateAddressTypes 

		multiple visit address error - for each permeatation?  writeToUsTypeId && !!primary.type_id && !!secondary.type_id && (this.addressFieldsNotEmpty(secondary) 
			|| secondary.postcode?.trim()) && // validate only if secondary has some fields entered
    			(primary.type_id !== writeToUsTypeId && secondary.type_id !== writeToUsTypeId) // at least 1 write address should exist

	^ show in addressValid block 


	Post api for addresses valid success

	post api fail 400

		Primary postcode invalid 

		secondary postcode invalid 
		
		postcode validation errors === 0, other failure besides postcodes 

		A non 400 error message (display update address error) 


	Success: 

		Gets addresses 

		Gets address types 

		getAddressTypesForSelect 

			primary address returns all 

			secondary address returns all matching allAddressTypes variable 

		
		

	
```

Reply:
I think the above are covered. The main one I could see missing was a test around 'address type' validation.